tests/scripts/all.sh

```diff
@@ -1899,14 +1899,20 @@ component_test_tls1_2_default_stream_cipher_only_use_psa () {
     msg "build: default with only stream cipher use psa"
 
     scripts/config.py set MBEDTLS_USE_PSA_CRYPTO
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
     # Disable AEAD (controlled by the presence of one of GCM_C, CCM_C, CHACHAPOLY_C)
+    scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CCM
+    scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_GCM
+    scripts/config.py -f $CRYPTO_CONFIG_H unset PSA_WANT_ALG_CHACHA20_POLY1305
+    # Note: The three unsets below are to be removed for Mbed TLS 4.0
     scripts/config.py unset MBEDTLS_GCM_C
     scripts/config.py unset MBEDTLS_CCM_C
     scripts/config.py unset MBEDTLS_CHACHAPOLY_C
     #Disable TLS 1.3 (as no AEAD)
     scripts/config.py unset MBEDTLS_SSL_PROTO_TLS1_3
-    # Disable CBC-legacy (controlled by MBEDTLS_CIPHER_MODE_CBC plus at least one block cipher (AES, ARIA, Camellia, DES))
-    scripts/config.py unset MBEDTLS_CIPHER_MODE_CBC
```

Comment:
I did not check completely but isn't the case that we must keep that one as well?

Reply:
I am a little bit confused by this actually and just want to clarify. Are you saying/suggesting that we need to keep this one unset so that we can fully turn off AEAD (since CBC is a mode that is supported by AEAD) and there is a not a PSA equivalent to `MBEDTLS_CIPHER_MODE_CBC` at this time?

I think that is what you're saying, and if so, I think I agree, so I have done this. Please let me know if I am wrong on this though.

Reply:
Looking in `config_adjust_legacy_from_psa.h` I see the following `defines`:
```
#if defined(MBEDTLS_CIPHER_MODE_CBC)
#define MBEDTLS_PSA_BUILTIN_ALG_CBC_NO_PADDING 1
#define PSA_WANT_ALG_CBC_NO_PADDING 1
```
Perhaps `PSA_WANT_ALG_CBC_NO_PADDING` can be used instead of `MBEDTLS_CIPHER_MODE_CBC` but I am not sure that these are quite an equivalent swap.

Reply:
This component is for stream ciphers only, so it should disable CBC in all forms.

Also, preexisting, I think this component should have `MBEDTLS_BLOCK_CIPHER_NO_DECRYPT` enabled. That option didn't exist when the component was created, but should have been added when we added the option.